### PR TITLE
[update] Troubleshooting section with the pg_dump error

### PR DESCRIPTION
**Pull request by Rajakavitha1**
# Description

Added the error message, cause, and workaround for the error when using pg_dump on self-hosted.

# Links

Fixes https://github.com/timescale/docs/issues/2226

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/about/latest/contribute-to-docs/)

# Review checklists

Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

*   [ ] Is the content technically accurate?
*   [ ] Is the content complete?
*   [ ] Is the content presented in a logical order?
*   [ ] Does the content use appropriate names for features and products?
*   [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

*   [ ] Is the content free from typos?
*   [ ] Does the content use plain English?
*   [ ] Does the content contain clear sections for concepts, tasks, and references?
*   [ ] Have any images been uploaded to the correct location, and are resolvable?
*   [ ] If the page index was updated, are redirects required
      and have they been implemented?
*   [ ] Have you checked the built version of this content?

### CODE REVIEW

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<!---
+* Use this format for writing troubleshooting sections:
+ - Cause: What causes the problem?
+ - Consequence: What does the user see when they hit this problem?
+ - Fix/Workaround: What can the user do to fix or work around the problem? Provide a "Resolving" Procedure if required.
+ - Result: When the user applies the fix, what is the result when the same action is applied?
+* Copy this comment at the top of every troubleshooting page
+-->
+ The `pg_dump` tool tries to acquire lock on the job_errors table but
+ the user doesn't have the required SELECT permission resulting in the error.
```
Comment (Loquacity) [inline _troubleshooting/self-hosted/pg_dump-permission-denied.md:23]:
```suggestion

 When the `pg_dump` tool tries to acquire a lock on the `job_errors` 
 table, if the user doesn't have the required SELECT permission, it 
 results in this error.
```

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+ A superuser needs to grant the necessary permission to the user who needs to
+ take the backup using the `pg_dump` tool.
```
Comment (Loquacity) [inline _troubleshooting/self-hosted/pg_dump-permission-denied.md:26]:
```suggestion
To resolve this issue, use a superuser account to grant the necessary 
permissions to the user requiring the `pg_dump` tool.
```

**Review by Loquacity — APPROVED** [review]

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: Permission denied for table job_errors when running `pg_dump`
```
Comment (lkshminarayanan) [inline _troubleshooting/self-hosted/pg_dump-permission-denied.md:2]: job_errors shows up as job _errors_ in the document. Can you add the backticks to job_errors as well?

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ - Result: When the user applies the fix, what is the result when the same action is applied?
+* Copy this comment at the top of every troubleshooting page
+-->
+
+ When the `pg_dump` tool tries to acquire a lock on the `job_errors` 
+ table, if the user doesn't have the required SELECT permission, it 
+ results in this error.
+
+To resolve this issue, use a superuser account to grant the necessary 
+permissions to the user requiring the `pg_dump` tool.
```
Comment (lkshminarayanan) [inline _troubleshooting/self-hosted/pg_dump-permission-denied.md:28]: Is it possible to include the exact grant command required in the document?

```sql
GRANT SELECT ON TABLE _timescaledb_internal.job_errors TO <testuser>;
```

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md (hunk 3 above)
Comment (Rajakavitha1) [reply _troubleshooting/self-hosted/pg_dump-permission-denied.md]: done!!!

File: _troubleshooting/self-hosted/pg_dump-permission-denied.md (hunk 4 above)
Comment (Rajakavitha1) [reply _troubleshooting/self-hosted/pg_dump-permission-denied.md]: Done!!!

**Review by lkshminarayanan — APPROVED** [review]
Thank you for updating this!

**Pull request merged**